addons/website_slides/views/slide_slide_views.xml

```diff
@@ -89,58 +129,38 @@
                                 <field name="description" placeholder="e.g. In this video, we'll give you the keys on how Odoo can help you to grow your business. At the end, we'll propose you a quiz to test your knowledge."/>
                             </page>
                             <page string="Additional Resources" name="external_links" >
-                                <group string="External Links">
-                                    <field name="link_ids" widget="one2many" nolabel="1">
-                                        <tree editable="top">
-                                            <field name="name"/>
-                                            <field name="link" widget="url" placeholder="e.g. https://www.odoo.com"/>
-                                        </tree>
-                                    </field>
-                                </group>
-                                <group string="Resources">
+                                <group>
                                     <field name="slide_resource_ids" widget="one2many" nolabel="1">
                                         <tree editable="top">
+                                            <field name="type"/>
                                             <field name="name"/>
-                                            <field name="data" string="Size" required="1"/>
+                                            <field name="data" attrs="{'readonly': [('type', '=', 'url')]}"/>
+                                            <field name="link" string="Link" attrs="{'readonly': [('type', '=', 'file')]}"/>
```

Comment:
Before your changes the link was required, could be grate to make it required if the type is url, what do you think ?

Reply:
Not a bad idea, and this way we keep the original idea for links.